src/create-glamorous.js

```diff
@@ -148,16 +174,54 @@ function createGlamorous(splitProps) {
       // set the displayName to something that's slightly more
       // helpful than `GlamorousComponent` :)
       displayName: displayName || `glamorous(${getDisplayName(comp)})`,
+      // these are props that should be applied to the component at render time
+      propsToApply,
     }
   }
+}
 
-  function when(comp, prop) {
-    return comp ? comp.concat(prop) : prop
-  }
+/**
+ * reduces the propsToApply given to a single props object
+ * @param {Array} propsToApply an array of propsToApply objects:
+ *   - object
+ *   - array of propsToApply items
+ *   - function that accepts the accumulated props and the context
+ * @param {Object} accumulator an object to apply props onto
+ * @param {Object} props the props that should ultimately take precedence
+ * @param {*} context the context object
+ * @return {Object} the reduced props
+ */
+function getPropsToApply(propsToApply, accumulator, props, context) {
```

Comment:
is this intended to be used elsewhere in the future? otherwise you could drop passing an empty object above and simplify this.

```
function getPropsToApply(propsToApply, props, context) {
    const accumulator = {}
```